### PR TITLE
[mono] Initial static ICU implementation for wasm

### PR DESCRIPTION
**Pull request by EgorBo**
Steps to try it:
clone ICU repo and build it via the following script:
https://github.com/kg/mono/blob/wasm-libicu/sdks/wasm/icu/build-icu.sh
set `export ICU_REPO=path/to/icu/repo`

build dotnet/runtime for wasm.

cc @kg

### CODE REVIEW

File: src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_internal.h [hunk 1]
```diff
@@ -280,3 +281,5 @@ FOR_ALL_ICU_FUNCTIONS
 #define usearch_getMatchedLength(...) usearch_getMatchedLength_ptr(__VA_ARGS__)
 #define usearch_last(...) usearch_last_ptr(__VA_ARGS__)
 #define usearch_openFromCollator(...) usearch_openFromCollator_ptr(__VA_ARGS__)
+
+#endif // !defined(ICU_STATIC_LINKING)
```
Comment (kg) [inline src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_internal.h:285]: isn't this !defined(STATIC_ICU)

**Comment by vargaz** [comment]
Why is this needed ? Why can't we just link icu into the final wasm executable ?

**Comment by EgorBo** [comment]
@vargaz I am not sure I understand what you mean, the __Internal changes will be removed once Thays PR for QCalls is merged. The rest is just to link static ICU files into the build
We don't have a separate `libSystem.Globalization.dylib` any more and the ICU shim was merged into runtime a while ago.

**Comment by EgorBo** [comment]
Closing due to https://github.com/dotnet/runtime/pull/37971

**Pull request closed without merging**